urllib3/connection.py

```diff
@@ -162,6 +162,27 @@ def connect(self):
         conn = self._new_conn()
         self._prepare_conn(conn)
 
+    def request_chunked(self, method, url, body=None, headers={}):
+        """ 
+        Alternative to the common request method, which sends the 
+        body with chunked encoding and not as one block 
+        """
+        header_names = set(k.lower() for k in headers)
+        self.putrequest(method, url, skip_accept_encoding=True)
+        for header, value in headers.items():
+            self.putheader(header, value)
+        if 'transfer-encoding' not in headers:
+            self.putheader('Transfer-Encoding', 'chunked')
```

Comment:
It's entirely allowed to have several different transfer encodings, and they're order-specific too: for example, `Transfer-Encoding: gzip, deflate, chunked` says that the originally representation was on-the-fly gzipped, which was then on-the-fly deflated, which was then chunk-encoded and sent, and the remote party needs to dechunk it, then inflate it, then gunzip it to get the original.

Like I said, it's *very* uncommon, but it is technically allowed.

Automatically setting identity is a bit different, though: which library does that? We should really stop, RFC 7230 deprecated it.